homeassistant/components/insteon_plm.py

```diff
@@ -26,95 +26,183 @@
 
-PLM_PLATFORMS = {
-    'binary_sensor': ['binary_sensor'],
-    'light': ['light'],
-    'switch': ['switch'],
-}
-
 
 @asyncio.coroutine
 def async_setup(hass, config):
     """Set up the connection to the PLM."""
     import insteonplm
 
+    ipdb = IPDB()
+
     conf = config[DOMAIN]
     port = conf.get(CONF_PORT)
-    overrides = conf.get(CONF_OVERRIDE)
+    overrides = conf.get(CONF_OVERRIDE, [])
 
     @callback
     def async_plm_new_device(device):
         """Detect device from transport to be delegated to platform."""
-        name = device.get('address')
-        address = device.get('address_hex')
-        capabilities = device.get('capabilities', [])
-
-        _LOGGER.info("New INSTEON PLM device: %s (%s) %r",
-                     name, address, capabilities)
-
-        loadlist = []
-        for platform in PLM_PLATFORMS:
-            caplist = PLM_PLATFORMS.get(platform)
-            for key in capabilities:
-                if key in caplist:
-                    loadlist.append(platform)
-
-        loadlist = sorted(set(loadlist))
-
-        for loadplatform in loadlist:
-            hass.async_add_job(
-                discovery.async_load_platform(
-                    hass, loadplatform, DOMAIN, discovered=[device],
-                    hass_config=config))
+        for state_key in device.states:
+            platform_info = ipdb[device.states[state_key]]
+            platform = platform_info.platform
+            if platform is not None:
+                _LOGGER.info("New INSTEON PLM device: %s (%s) %s",
+                             device.address,
+                             device.states[state_key].name,
+                             platform)
+
+                hass.async_add_job(
+                    discovery.async_load_platform(
+                        hass, platform, DOMAIN,
+                        discovered={'address': device.address.hex,
+                                    'state_key': state_key},
+                        hass_config=config))
 
     _LOGGER.info("Looking for PLM on %s", port)
-    plm = yield from insteonplm.Connection.create(device=port, loop=hass.loop)
+    conn = yield from insteonplm.Connection.create(
+        device=port,
+        loop=hass.loop,
+        workdir=hass.config.config_dir)
+
+    plm = conn.protocol
 
-    for device in overrides:
+    for device_override in overrides:
         #
         # Override the device default capabilities for a specific address
         #
-        if isinstance(device['platform'], list):
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', device['platform'])
-        else:
-            plm.protocol.devices.add_override(
-                device['address'], 'capabilities', [device['platform']])
+        address = device_override.get('address')
+        if address is not None:
+            if device_override.get('cat', False):
+                plm.devices.add_override(address,
```

Comment:
Maybe do this with a `for` loop, iterating over the four optional keys of the override. If there is a value for the key in the override, add the override.